### PR TITLE
Support managing the creation and deletion of users for a bucket

### PR DESCRIPTION
**Pull request by fishnix**

### CODE REVIEW

File: iam/user.go [hunk 1]
```diff
@@ -0,0 +1,360 @@
...
+// ListUserGroups returns a list of groups that a user belongs to
+func (i *IAM) ListUserGroups(ctx context.Context, input *iam.ListGroupsForUserInput) ([]*iam.Group, error) {
+	groups := []*iam.Group{}
+
+	if input == nil || aws.StringValue(input.UserName) == "" {
+		return groups, apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+
+	truncated := true
+	for truncated {
```
Comment (tenyo) [inline iam/user.go]: Is this for pagination?

File: iam/user_test.go [hunk 2]
```diff
@@ -0,0 +1,931 @@
...
+	i.Service.(*mockIAMClient).err = errors.New("things blowing up!")
+	_, err = i.ListUserGroups(context.TODO(), &iam.ListGroupsForUserInput{UserName: aws.String("testuser")})
+	if aerr, ok := err.(apierror.Error); ok {
+		if aerr.Code != apierror.ErrInternalError {
+			t.Errorf("expected error code %s, got: %s", apierror.ErrInternalError, aerr.Code)
+		}
+	} else {
+		t.Errorf("expected apierror.Error, got: %s", reflect.TypeOf(err).String())
+	}
+}
```
Comment (tenyo) [inline iam/user_test.go]: 😮 Test coverage 💯 😆

File: README.md [hunk 3]
```diff
@@ -147,6 +147,39 @@ DELETE `/v1/s3/{account}/buckets/{bucket}
 | **409 Conflict**              | bucket is not empty             |
 | **500 Internal Server Error** | a server error occurred         |
 
+### Create a bucket user
+
+POST `/v1/s3/{account}/buckets/{bucket}/users
```
Comment (tenyo) [inline README.md]: Let's also add an example of what this is returning besides the code. That will be very helpful when trying to use the API.

File: iam/user.go (hunk 1 above)
Comment (fishnix) [reply iam/user.go]: yeah

File: README.md (hunk 3 above)
Comment (fishnix) [reply README.md]: 👍

**Pull request merged**